torch/nn/modules/lazy.py

```diff
@@ -159,8 +164,8 @@ class LazyModuleMixin:
                             0.2479,  0.1091]])),
                  ('fc2.bias', tensor([0.0019]))])
 
-    Note, however, that lazy modules cannot validate that the shape of parameters they load is correct.
-
+    Note, however, that the loaded parameters will not be replaced when doing a "dry run" if they are initialized
+    and the input shapes are different. This prevents using initialized modules in different contexts.
```

Comment:
I think whether `the input shapes are different` is not related to the lazy behavior.
How about just saying:
```
Note, however, that the loaded parameters will not be replaced when doing a "dry run" if they are already initialized.
```